src/Util/AdminObjectAclData.php

```diff
@@ -77,6 +77,9 @@ class AdminObjectAclData
      */
     protected $maskBuilderClass;
 
+    /** @var AclSecurityHandlerInterface */
+    private $securityHandler;
```

Comment:
This property is not needed.

You didn't like my proposal ?
In the construct, 
```
if (!$admin->isAclEnabled()) {
   throw new BadArgument('The admin should enable the acl');
}
```

Then 
```
public function getSecurityHandler()
{
   $securityHandler = $this->admin->getSecurityHandler();
   assert($securityHandler instanceof AclSecurityHandlerInterface);

   return $securityHandler;
}

Then 
```
public function getPermissions()
{
    return $this->getSecurityHandler()->getPermissions();
}
```
And same for others.

Reply:
I like it :)

I didn't get it first time. Now I get it.